source/agora/cli/multi/main.d

```diff
@@ -0,0 +1,251 @@
+/*******************************************************************************
+
+    Entry point for the Agora multi
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.cli.multi.main;
+
+import agora.api.FullNode;
+
+import agora.common.Config;
+import agora.node.Node;
+import agora.utils.Log;
+
+import vibe.core.core;
+import vibe.http.router;
+import vibe.http.server;
+import vibe.web.rest;
+
+import ocean.util.log.Logger;
+
+import std.file;
+import std.format;
+import std.getopt;
+import std.path;
+import std.stdio;
+
+import core.thread;
+
+mixin AddLogger!();
+
+/// Application entry point
+private int main (string[] args)
+{
+    if (args.length < 3)
+    {
+        writeln("usage: agora-multi path file1.yaml file2.yaml file3.yaml ...");
+    }
+
+    string path = args[1];
+    string[] files;
+
+    foreach (idx; 2..args.length)
+        files ~= buildNormalizedPath(path, args[idx]);
+
+    NodeLauncher[] launchers;
+
+    foreach (ref file; files)
+    {
+        try
+        {
+            CommandLine cmdln;
+            cmdln.config_path = file;
+            auto config = parseConfigFile(cmdln);
+
+            auto launcher = new NodeLauncher();
+
+            //launcher.startThread(config);
+            launcher.startTask(config);
+
+            launchers ~= launcher;
+
+            testConection(format("http://%s:%d", config.node.address, config.node.port));
+        }
+        catch (Exception ex)
+        {
+            writefln("Failed to parse config file '%s'. Error: %s",
+                file, ex.message);
+            return 1;
+        }
+    }
+
+    scope (exit)
+    {
+        foreach (ref elem; launchers)
+            if (elem !is null)
+                elem.stop();
+
+        exitEventLoop();
+    }
+
+    return runEventLoop();
+}
+
+
+/*******************************************************************************
+
+    Test the connection to the node.
+
+    Params:
+      address = An address of the node
+
+*******************************************************************************/
+
+private void testConection (string address)
+{
+    import core.time;
+    runTask({
```

Comment:
If you wrap the whole function in `runTask`, just take it out of `testConnection` and put it in the caller:
```D
runTask(&testConnection, address);
```